alchemist/alchemist-smartcam/src/main/kotlin/it/unibo/alchemist/model/implementations/actions/ExtractClosestVisibleNodeAtDistance.kt

```diff
@@ -0,0 +1,58 @@
+package it.unibo.alchemist.model.implementations.actions
+
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings
+import it.unibo.alchemist.model.implementations.positions.Euclidean2DPosition
+import it.unibo.alchemist.model.interfaces.Context
+import it.unibo.alchemist.model.interfaces.Molecule
+import it.unibo.alchemist.model.interfaces.Node
+import it.unibo.alchemist.model.interfaces.Reaction
+import it.unibo.alchemist.model.interfaces.VisibleNode
+import it.unibo.alchemist.model.interfaces.environments.EuclideanPhysics2DEnvironment
+import it.unibo.alchemist.model.smartcam.offsetPositionAtDistance
+import kotlin.math.min
+
+/**
+ * Extracts the closest [VisibleNode] to the position at the given distance from the given node's position
+ * in the direction of the node's heading.
+ * @constructor
+ * [visionMolecule] is expected to contain a list of [VisibleNode].
+ * If there are no visible nodes then [targetMolecule] will be removed from [node],
+ * otherwise the VisibleNode's position will be inserted in [targetMolecule].
+ */
+class ExtractClosestVisibleNodeAtDistance(
+    node: Node<Any>,
+    private val env: EuclideanPhysics2DEnvironment<Any>,
+    private val distance: Double,
+    private val visionMolecule: Molecule,
+    private val targetMolecule: Molecule
+) : AbstractAction<Any>(node) {
+    override fun cloneAction(n: Node<Any>, r: Reaction<Any>) =
+        ExtractClosestVisibleNodeAtDistance(n, env, distance, visionMolecule, targetMolecule)
+
+    @SuppressFBWarnings("RCN_REDUNDANT_NULLCHECK_WOULD_HAVE_BEEN_A_NPE")
```

Comment:
Remove this suppression and disable the check for Kotlin files. Too many false positives

Reply:
Done